menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -44,15 +48,27 @@ class SchemaController @Autowired() (
   @ResponseStatus(HttpStatus.CREATED)
   def handleFileUpload(@AuthenticationPrincipal principal: UserDetails,
                        @RequestParam file: MultipartFile,
-                       @RequestParam version: Int, @RequestParam name: String): CompletableFuture[_] = {
+                       @RequestParam version: Int,
+                       @RequestParam name: String,
+                       @RequestParam format: Optional[String]): CompletableFuture[_] = {
+    val origFormat: Option[String] = format
+    val fileContent = new String(file.getBytes)
+
+    val sparkStruct = origFormat match {
+      case Some("struct") | Some("") | None => sparkMenasConvertor.convertAnyToStructType(fileContent)
+      case Some("copybook") =>
+        val parsedSchema = CopybookParser.parseTree(fileContent)
+        val cobolSchema = new CobolSchema(parsedSchema, SchemaRetentionPolicy.CollapseRoot, false)
+        cobolSchema.getSparkSchema
+      case Some(x) => throw new UnsupportedOperationException(s"$x is not a recognized schema format. Menas currently supports: 'struct' and 'copybook'")
+    }
```

Comment:
Doesn't this `match` miss the `case None` code branch? If it's an `Option[String]` I would expect it to be possible `None` would come in.

Reply:
It doesn't. Read again

Reply:
My bad, sorry. 😊